iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/CompoundTextRange.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.tree.impl;
+
+import java.util.List;
+import org.sonar.iac.common.api.tree.impl.TextRange;
+import org.sonar.iac.common.api.tree.impl.TextRanges;
+
+public class CompoundTextRange extends TextRange {
+  private final List<TextRange> textRanges;
+
+  public CompoundTextRange(List<TextRange> textRanges) {
+    super(textRanges.get(0).start(), textRanges.get(textRanges.size() - 1).end());
+    this.textRanges = textRanges;
```

Comment:
SC reports java:S2384 here (Store a copy of "textRanges".).
IMO It is "won't fix". This rule only applies to immutable objects, making a copy of the collection each time when passed to an object/method is in our case overkill.